### PR TITLE
Feature/ereporting refactor

### PR DESCRIPTION
**Pull request by jianmingtu**

### CODE REVIEW

File: icon-two-common-models/src/main/resources/xsdSchemas/e-reporting-models-1.xsd [hunk 1]
```diff
@@ -183,6 +139,7 @@
 
     <xsd:complexType name="question">
         <xsd:sequence>
+            <xsd:element name="QuestionId" nillable="true" type="xsd:string"/>
```
Comment (sylar-zeyangz) [inline icon-two-common-models/src/main/resources/xsdSchemas/e-reporting-models-1.xsd:142]: why cap?

File: icon-two-common-models/src/main/resources/xsdSchemas/e-reporting-models-1.xsd [hunk 2]
```diff
@@ -110,20 +87,13 @@
     <xsd:element name="setMessageDateDocument">
         <xsd:complexType>
             <xsd:sequence>
-                <xsd:element name="XMLString" nillable="true" type="tns:AppointmentMessageOuter"/>
-                <xsd:element name="UserTokenString" nillable="true" type="tns:userTokenOuter"/>
+                <xsd:element name="AppointmentMessage" nillable="true" type="tns:AppointmentMessage"/>
```
[the remark this answers is not in the document]
Comment (jianmingtu) [reply icon-two-common-models/src/main/resources/xsdSchemas/e-reporting-models-1.xsd]: done

File: icon-two-common-models/src/main/resources/xsdSchemas/e-reporting-models-1.xsd (hunk 1 above)
Comment (jianmingtu) [reply icon-two-common-models/src/main/resources/xsdSchemas/e-reporting-models-1.xsd]: done

File: icon-two-common-application-1-1/src/main/java/ca/bc/gov/open/icon/controllers/MessageController.java [hunk 3]
```diff
@@ -125,37 +128,35 @@ public GetMessageResponse getMessage(@RequestPayload GetMessage getMessage)
...
+                        setMessageDate.getUserTokenString(),
+                        new ca.bc.gov.open.icon.ereporting.UserToken()));
 
         UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(host + "message/date");
         HttpEntity<SetMessageDateDocument> payload =
                 new HttpEntity<>(setMessageDateDocument, new HttpHeaders());
 
         try {
-            HttpEntity<AppointmentMessage> resp =
+            HttpEntity<Map<String, String>> resp =
```
[the remark this answers is not in the document]
Comment (jianmingtu) [reply icon-two-common-application-1-1/src/main/java/ca/bc/gov/open/icon/controllers/MessageController.java]: done

File: icon-two-common-application-1-1/src/test/java/ca/bc/gov/open/icon/MessageControllerTests.java [hunk 4]
```diff
@@ -107,24 +106,20 @@ public void testGetMessage() throws JsonProcessingException {
...
     }
 
     @Test
     public void testSetMessageDate() throws JsonProcessingException {
         var req = new SetMessageDate();
+        req.setXMLString(
+                "<AppointmentMessage>\n" + "    <csNum>1</csNum>\n" + "</AppointmentMessage> ");
 
-        var AppointmentMessageOuter = new AppointmentMessageOuter();
         var AppointmentMessage = new AppointmentMessage();
```
Comment (sylar-zeyangz) [inline icon-two-common-application-1-1/src/test/java/ca/bc/gov/open/icon/MessageControllerTests.java:119]: please follow the naming standard

**Review by sylar-zeyangz — CHANGES_REQUESTED** [review]

**Review by sylar-zeyangz — APPROVED** [review]

File: icon-two-common-application-1-1/src/main/java/ca/bc/gov/open/icon/controllers/MessageController.java [hunk 5]
```diff
@@ -93,19 +99,14 @@ public GetMessageResponse getMessage(@RequestPayload GetMessage getMessage)
                             HttpMethod.POST,
                             payload,
                             AppointmentMessage.class);
+
+            GetMessageResponse getMessageResponse = new GetMessageResponse();
+            getMessageResponse.setXMLString(XMLUtilities.serializeXmlStr(resp.getBody()));
```
Comment (sylar-zeyangz) [inline icon-two-common-application-1-1/src/main/java/ca/bc/gov/open/icon/controllers/MessageController.java:104]: wrong

**Review by sylar-zeyangz — CHANGES_REQUESTED** [review]

File: icon-two-common-application-1-1/src/main/java/ca/bc/gov/open/icon/controllers/MessageController.java (hunk 5 above)
Comment (jianmingtu) [reply icon-two-common-application-1-1/src/main/java/ca/bc/gov/open/icon/controllers/MessageController.java]: done

**Review by sylar-zeyangz — APPROVED** [review]

**Pull request merged**